nix/modules/marlowe-playground.nix

```diff
@@ -0,0 +1,93 @@
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib) types mkOption mkIf;
+  cfg = config.services.marlowe-playground;
+
+  killallz3 = pkgs.writeScriptBin "killallz3" ''
+    kill -9 $(ps aux | grep z3 | grep -v grep | awk '{print $2}')
+  '';
+
+in
+{
+  options.services.marlowe-playground = {
+    enable = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        If enabled the marlowe-playground server will be started.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 4001;
+      description = ''
+        Port the marlowe-playground server should bind to.
+      '';
+    };
+
+    frontendURL = mkOption {
+      type = types.str;
+      default = "http://localhost:4000";
+      description = ''
+        URL where the plutus playground is served.
+      '';
+    };
+
+    githubCallbackPath = mkOption {
+      type = types.str;
+      default = "/#/gh-oauth-cb";
+      description = ''
+        The github callback path
+      '';
+    };
+
+    playground-server-package = mkOption {
+      type = types.package;
+      description = ''
+        marlowe playground package to execute.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.services.marlowe-playground = {
+      after = [ "network.target" ];
+      wantedBy = [ "nginx.service" ];
+      before = [ "nginx.service" ];
+
+      serviceConfig = {
+        # runtime behavior
+        TimeoutStartSec = 5;
+        TimeoutStopSec = 5;
+        Restart = "always";
+
+        # sane defaults for security
+        DynamicUser = true;
+        ProtectKernelTunables = true;
+        ProtectControlGroups = true;
+        ProtectKernelModules = true;
+        PrivateDevices = true;
+        SystemCallArchitectures = "native";
+
+      };
+
+      path = [ pkgs.z3 killallz3 ];
+      script = ''
+        if [ -f /var/lib/playgrounds/marlowe.env ]; then
+          echo "Loading environment config from '/var/lib/playgrounds/marlowe.env'"
```

Comment:
Should this instead be an option? Either the path, or even the file iteslf. Or does it need to be discovered at runtime?

Reply:
Also we don't do anything with this path? I guess the server looks for it?

Reply:
This path will contain environment variables that provide secrets. They are going to be uploaded by `morph` during deployment. The service does however also work without these secrets hence they are optional here. The service gets restarted by morph after they have been uploaded.